From: Matthias Brugger <mbrugger@suse.com>
Date: Tue, 20 Mar 2018 13:56:21 +0000
Subject: arm64: fix smccc compilation
Git-commit: 82bcd087029f6056506ea929f11af02622230901 (partial)
Patch-mainline: v4.11-rc1
References: bsc#1068032
Backport-note: Only contains fixing arm-smccc.h -jeffm

ARM_SMCC_CALL_VAL is a assembly macro. Fix compilation of
arm-smccc.h by setting ifdefs as in mainline.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>---
 arch/arm64/kernel/smccc-call.S |    1 +
 include/linux/arm-smccc.h      |    9 ++++++---
 2 files changed, 7 insertions(+), 3 deletions(-)

--- a/arch/arm64/kernel/smccc-call.S
+++ b/arch/arm64/kernel/smccc-call.S
@@ -12,6 +12,7 @@
  *
  */
 #include <linux/linkage.h>
+#include <linux/arm-smccc.h>
 #include <asm/asm-offsets.h>
 
 	.macro SMCCC instr
--- a/include/linux/arm-smccc.h
+++ b/include/linux/arm-smccc.h
@@ -14,9 +14,6 @@
 #ifndef __LINUX_ARM_SMCCC_H
 #define __LINUX_ARM_SMCCC_H
 
-#include <linux/linkage.h>
-#include <linux/types.h>
-
 /*
  * This file provides common defines for ARM SMC Calling Convention as
  * specified in
@@ -80,6 +77,11 @@
 #define ARM_SMCCC_OWNER_TRUSTED_OS	50
 #define ARM_SMCCC_OWNER_TRUSTED_OS_END	63
 
+#ifndef __ASSEMBLY__
+
+#include <linux/linkage.h>
+#include <linux/types.h>
+
 /**
  * struct arm_smccc_res - Result from SMC/HVC call
  * @a0-a3 result values from registers 0 to 3
@@ -262,4 +264,5 @@ asmlinkage void arm_smccc_hvc(unsigned l
 			unsigned long a5, unsigned long a6, unsigned long a7,
 			struct arm_smccc_res *res);
 
+#endif /*__ASSEMBLY__*/
 #endif /*__LINUX_ARM_SMCCC_H*/
